File: src/windows/AmbientContextMcp.Core.Tests/PollSummaryModeTests.cs
using AmbientContextMcp.Core.Hub;
using AmbientContextMcp.Core.Models;
using Xunit;

namespace AmbientContextMcp.Core.Tests;

public class PollSummaryModeTests
{
    [Fact]
    public void Default_poll_request_includes_payload()
    {
        var hub = LocalContextHubTestFactory.CreateInMemory();
        var observedAt = DateTimeOffset.UtcNow;

        hub.Ingest(BuildSingleEventSnapshot(observedAt));

        var poll = hub.PollEvents(new LocalContextPollRequest
        {
            ClientId = "test",
            Scopes = ["context.high:read"],
            Since = observedAt.AddMinutes(-1)
        });

        var ev = Assert.Single(poll.Events);
        Assert.Equal("Imagine", ev.Payload["title"]);
        Assert.Equal("Chrome", ev.Payload["source_app"]);
    }

    [Fact]
    public void Summary_mode_strips_payload_and_payload_sensitivity()
    {
        var hub = LocalContextHubTestFactory.CreateInMemory();
        var observedAt = DateTimeOffset.UtcNow;

        hub.Ingest(BuildSingleEventSnapshot(observedAt));

        var poll = hub.PollEvents(new LocalContextPollRequest
        {
            ClientId = "test",
            Scopes = ["context.high:read"],
            Since = observedAt.AddMinutes(-1),
            IncludePayload = false
        });

        var ev = Assert.Single(poll.Events);
        Assert.Empty(ev.Payload);
        Assert.Empty(ev.PayloadSensitivity);
    }

    [Fact]
    public void Summary_mode_preserves_top_level_metadata()
    {
        var hub = LocalContextHubTestFactory.CreateInMemory();
        var observedAt = DateTimeOffset.UtcNow;

        hub.Ingest(BuildSingleEventSnapshot(observedAt));

        var poll = hub.PollEvents(new LocalContextPollRequest
        {
            ClientId = "test",
            Scopes = ["context.high:read"],
            Since = observedAt.AddMinutes(-1),
            IncludePayload = false
        });

        var ev = Assert.Single(poll.Events);
        Assert.False(string.IsNullOrEmpty(ev.Id));
        Assert.Equal("media_session_changed", ev.Name);
        Assert.Equal("Imagine", ev.Value);
        Assert.Equal("medium", ev.Sensitivity);
        Assert.Equal("high", ev.MaxFieldSensitivity);
    }

    private static AmbientContextSnapshot BuildSingleEventSnapshot(DateTimeOffset observedAt) =>
        new()
        {
            ObservedAt = observedAt,
            PrivacyClassifications =
            [
                new() { Path = "events.media_session_changed", Sensitivity = "medium", DefaultTransmit = false },
                new() { Path = "events.media_session_changed.title", Sensitivity = "high", DefaultTransmit = false }
            ],
            OutboundEvents =
            [
                new()
                {
                    ObservedAt = observedAt,
                    Name = "media_session_changed",
                    Value = "Imagine",
                    Payload = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
                    {
                        { "title", "Imagine" },
                        { "source_app", "Chrome" }
                    },
                    Sensitivity = "medium"
                }
            ]
        };
}

File: src/macos/Tests/AmbientContextCoreTests/PollSummaryModeTests.swift
import Foundation
import Testing
@testable import AmbientContextCore

@Suite("PollSummaryMode")
struct PollSummaryModeTests {
    private static func buildSingleEventSnapshot(observedAt: Date) -> AmbientContextSnapshot {
        AmbientContextSnapshot(
            observedAt: observedAt,
            outboundEvents: [
                AmbientOutboundEvent(
                    observedAt: observedAt,
                    name: "media_session_changed",
                    value: "Imagine",
                    payload: ["title": "Imagine", "source_app": "Chrome"],
                    sensitivity: "medium")
            ],
            privacyClassifications: [
                PrivacyClassification(path: "events.media_session_changed", sensitivity: "medium", defaultTransmit: false),
                PrivacyClassification(path: "events.media_session_changed.title", sensitivity: "high", defaultTransmit: false)
            ])
    }

    @Test("Default_poll_request_includes_payload")
    func defaultPollRequestIncludesPayload() throws {
        let hub = LocalContextHubTestFactory.createInMemory()
        let observedAt = Date()

        hub.ingest(Self.buildSingleEventSnapshot(observedAt: observedAt))

        let poll = hub.pollEvents(LocalContextPollRequest(
            clientId: "test",
            scopes: ["context.high:read"],
            since: observedAt.addingTimeInterval(-60)))

        #expect(poll.events.count == 1)
        let event = try #require(poll.events.first)
        #expect(event.payload["title"] == "Imagine")
        #expect(event.payload["source_app"] == "Chrome")
    }

    @Test("Summary_mode_strips_payload_and_payload_sensitivity")
    func summaryModeStripsPayloadAndPayloadSensitivity() throws {
        let hub = LocalContextHubTestFactory.createInMemory()
        let observedAt = Date()

        hub.ingest(Self.buildSingleEventSnapshot(observedAt: observedAt))

        let poll = hub.pollEvents(LocalContextPollRequest(
            clientId: "test",
            scopes: ["context.high:read"],
            since: observedAt.addingTimeInterval(-60),
            includePayload: false))

        #expect(poll.events.count == 1)
        let event = try #require(poll.events.first)
        #expect(event.payload.isEmpty)
        #expect(event.payloadSensitivity.isEmpty)
    }

    @Test("Summary_mode_preserves_top_level_metadata")
    func summaryModePreservesTopLevelMetadata() throws {
        let hub = LocalContextHubTestFactory.createInMemory()
        let observedAt = Date()

        hub.ingest(Self.buildSingleEventSnapshot(observedAt: observedAt))

        let poll = hub.pollEvents(LocalContextPollRequest(
            clientId: "test",
            scopes: ["context.high:read"],
            since: observedAt.addingTimeInterval(-60),
            includePayload: false))

        #expect(poll.events.count == 1)
        let event = try #require(poll.events.first)
        #expect(!event.id.isEmpty)
        #expect(event.name == "media_session_changed")
        #expect(event.value == "Imagine")
        #expect(event.sensitivity == "medium")
        #expect(event.maxFieldSensitivity == "high")
    }
}
